From d71491b7f0b0ae9d940435ae9ea7edcb9d28aa5e Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Sat, 14 May 2022 22:57:53 +0800
Subject: [PATCH 52/58] Revert "drm/bridge: Fix free wrong object in
 sii8620_init_rcp_input_dev"

This reverts commit d8db734df6e62c5cff6d7a3a1936f4831fac8bfc.
---
 drivers/gpu/drm/bridge/sil-sii8620.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/sil-sii8620.c b/drivers/gpu/drm/bridge/sil-sii8620.c
index ec7745c31da07..843265d7f1b12 100644
--- a/drivers/gpu/drm/bridge/sil-sii8620.c
+++ b/drivers/gpu/drm/bridge/sil-sii8620.c
@@ -2120,7 +2120,7 @@ static void sii8620_init_rcp_input_dev(struct sii8620 *ctx)
 	if (ret) {
 		dev_err(ctx->dev, "Failed to register RC device\n");
 		ctx->error = ret;
-		rc_free_device(rc_dev);
+		rc_free_device(ctx->rc_dev);
 		return;
 	}
 	ctx->rc_dev = rc_dev;
-- 
2.27.0

